src/components/hooks/usePreviousActivities.js

```diff
@@ -0,0 +1,67 @@
+import {useContext, useEffect, useRef, useState} from 'react';
+import {Observable} from 'rxjs';
+import {flatMap} from 'rxjs/operators';
+
+import {AdapterContext} from '../../components/';
+
+/**
+ * Callback function to execute once data has been fetched.
+ *
+ * @callback previousActivitiesCallback
+ * @param {Array.<string|ActivityDate>} previousActivities Array of previous activities fetched
+ * @returns undefined
+ */
+
+/**
+ * Custom hook that returns a reference to the DOM element of the last activity.
+ * Requests previous activity data for the given room, and executes
+ * the given callback with that data.
+ * It completes once there is enough data to fill the viewport of the
+ * parent container, finishing by scrolling into view the last activity.
+ *
+ * @param {string} roomID  ID of the room for which to load data.
+ * @param {object} elementRef  DOM reference of the parent component.
+ * @param {previousActivitiesCallback} callback  Callback to execute once data has been fetched.
+ * @returns {object}  DOM reference to the the last activity.
+ */
+export default function usePreviousActivities(roomID, elementRef, callback) {
+  const lastElementRef = useRef(null);
+  const {roomsAdapter} = useContext(AdapterContext);
+  const [scrollToLastActivity, setScrollToLastActivity] = useState(true);
+  const [lastActivityBottom, setLastActivityBottom] = useState(0);
+
+  useEffect(() => {
+    const activityStreamBoundaries = elementRef.current.getBoundingClientRect();
+    const lastActivityBoundaries = lastElementRef.current.getBoundingClientRect();
+
+    // If the bottom position of the last activity is less than
+    // the bottom position of the container that means that the last
+    // activity is visible and there is not enough content to overflow
+    // the container
```

Comment:
Does this mean that if the window is resized to be bigger it will automatically load more activities?

Reply:
It should adjust how much content it loads based on the size of the window, but once it has scrolled to the last activity - meaning it ran once - it will not load content again.